packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -517,7 +525,11 @@ export default class NewVm extends BaseComponent {
       return this._reset()
     }
 
-    this._setState({ template })
+    const { pathname } = this.props.location
+    this.context.router.push({
+      pathname,
+      query: template && { pool: template.$pool, template: template.id },
+    })
```

Comment:
With the `componentDidUpdate` implementation, `SelectVmTemplate`'s `onChange` should only set the URL query. Also, make sure you're not overriding any other potential query param:
```js
_onChangeTemplate = template => {
  const { pathname, query } = this.props.location
  this.context.router.push({
    pathname,
    query: { ...query, template: template && template.id },
  })
}
```
and leave:
```js
_initTemplate = template => {
  if (!template) {
    return this._reset()
  }

  const storeState = store.getState()
  ...
}
```